service/dap/server.go

```diff
@@ -1915,6 +1921,16 @@ func getNamedVariableCount(v *proc.Variable) int {
 func (s *Server) metadataToDAPVariables(v *fullyQualifiedVariable) ([]dap.Variable, error) {
 	var children []dap.Variable
 
+	switch v.Kind {
+	case reflect.Map:
+		children = append(children, dap.Variable{
```

Comment:
Shouldn't you be checking that number of children is not 0?

Reply:
Yes I will explicitly check this here. If it has no children, then the parent variable will not have a variables reference, so the children are never requested.

Reply:
Shouldn't ListOfBytesOrRunes use v.Len as well instead of length of children (which might fail to load although unlikely)?

Reply:
Another thought is to just check Children in the outer function once for all types. It will be more obvious that metadata is skipped for empty child-less variables.

Reply:
These are actually two different scenarios. For ListOfBytesOrRunes, we need to have an actual child to inspect to determine if it is the write kind of slice, but with a map, even if it isn't loaded at the time, we can still show the length.

We could attempt to load the slice if its Len > 0 but len(Children) == 0, but I'm not sure if thats worthwhile.

Reply:
Ah, good point.